File: 443-string-compression/solution.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        resStr = ""
        groupLetter = ''
        groupSize = 0
        for idx, i in enumerate(chars):
            if i == groupLetter:
                groupSize += 1
            if groupLetter == '':
                groupLetter = i
                groupSize = 1
            if idx == len(chars)-1 or chars[idx+1] != i:
                resStr += i
                if groupSize > 1:
                    resStr += str(groupSize)
                groupLetter = ''
                groupSize = 0
        chars[:] = ''.join(resStr)
        return len(chars)

File: 443-string-compression/solution.go
package main

import (
	"strconv"
)

func compress(chars []byte) int {
	resStr := ""
	var groupLetter *byte = nil
	var groupCount int = 0
	for idx, letter := range chars {
		if groupLetter != nil && letter == *groupLetter {
			groupCount += 1
		}
		if groupLetter == nil {
			groupLetter = &letter
			groupCount = 1
		}
		if idx == len(chars)-1 || chars[idx+1] != letter {
			resStr += string(*groupLetter)
			if groupCount > 1 {
				resStr += strconv.Itoa(groupCount)
			}
			groupLetter = nil
			groupCount = 0
		}
	}
	resBytes := []byte(resStr)
	copy(chars, resBytes)
	return len(resBytes)
}
